src/GraphBuilder.js

```diff
@@ -11,23 +12,80 @@ export default class GraphBuilder {
    */
   constructor(state) {
     this.state = state
+    this.dataset = rdf.dataset()
   }
 
   /**
    * @return {Graph} an RDF graph that represents the data in the state
    */
   get graph() {
-    const dataset = rdf.dataset()
+    const baseURI = rdf.namedNode('')
 
+    // Is there ever more than one base node?
     for (const key in this.state) {
-      /*
-       * TODO: this is just a simple statement insert to show this works. We'll
-       *       want to add more here.
-       */
-      dataset.add(rdf.quad(rdf.namedNode(''),
-        rdf.namedNode('http://www.w3.org/1999/02/22-rdf-syntax-ns#type'),
-        rdf.literal(key)))
+      this.buildTriplesForNode(baseURI, rdf.literal(key), this.state[key])
     }
-    return dataset
+    return this.dataset
+  }
+
+  /**
+   * @param {rdf.Term} baseURI
+   * @param {Object} value looks something like this:
+   *   { 'resourceTemplate:bf2:Note':
+   *     { '67Bm64T0p2s': { 'http://www.w3.org/2000/01/rdf-schema#label': [Object] } } }
+   */
+  buildTriplesForNestedObject(baseURI, value) {
+    // Is there ever more than one base node?
+    for (const key in value) {
+      const predicateListKey = Object.keys(value[key])[0]
+
+      this.buildTriplesForNode(baseURI, rdf.literal(key), value[key][predicateListKey])
+    }
+  }
+
+  /**
+   * @param {rdf.Term} baseURI
+   * @param {rdf.Term} type
+   * @param {Array.<Object>}
+   */
+  buildTriplesForNode(baseURI, type, predicateList) {
+    this.addTypeTriple(baseURI, type)
+
+    // Now add all the other properties
+    for (const predicate in predicateList) {
+      const value = predicateList[predicate]
+
+      if (_.isEmpty(value)) {
+        continue
+      }
+
+      if (value.items) {
+        for (const item of value.items) {
+          if (item.uri) {
+            this.dataset.add(rdf.quad(baseURI, rdf.namedNode(predicate), rdf.namedNode(item.uri)))
```

Comment:
Why?  What if the label changes upstream?  Should this be considered as a different ticket?

Reply:
It's a Bibframe convention to add label if available: http://www.loc.gov/bibframe/docs/pdf/bf2-conventions-march2017.pdf

Reply:
I'm not reading that the same as you.  It says "URI or label or both". It does not say "SHOULD" or "MUST" and there are definite trade offs to be made.  Since this seems to follow the recommendation by my reading, maybe we need a dev team discussion about it.